x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -317,14 +315,17 @@ protected long ramBytesUsedCompleted() {
 
     private void addMemory(long bytes, String label) {
         circuitBreaker.addEstimateBytesAndMaybeBreak(bytes, label);
-        totalRamBytesUsed += bytes;
+        if (CB_COMPLETED_LABEL.equals(label)) {
+            prevRamBytesUsedCompleted += bytes;
+        } else {
+            prevRamBytesUsedInFlight += bytes;
+        }
```

Comment:
This isn't needed, as the values are assigned after the function call to the actual memory consumption.

Reply:
Good point, it's completely redundant. We don't even need an addMemory() method actually